src/gordon_gcp/plugins/janitor/reconciler.py

```diff
@@ -264,6 +291,10 @@ def _parse_rrset_message(self, message):
         missing_rrsets = [
             rs for rs in desired_rrsets if rs not in actual_rrsets
         ]
+        # don't try to add root SOA/NS records
+        missing_rrsets = GDNSReconciler._remove_soa_and_root_ns(
```

Comment:
i'm wondering if these calls should live outside of `validate_rrsets_by_zone`, as they don't really relate to validating the rrsets by zone. perhaps just in the run method before calling `validate_rrsets_by_zone`?

Reply:
I mean, that's kind of a problem with the function to begin with, since it publishes the changes, not just validates them.  I can split it up, but I think in that case the filtering should be in the validator and the publish_change_messages calls should be in run().